cpp/test/neighbors/tiled_knn.cu

```diff
@@ -209,6 +213,72 @@ class TiledKNNTest : public ::testing::TestWithParam<TiledKNNInputs> {
                                                          float(0.001),
                                                          stream_,
                                                          true));
+      // also test out the batch api. First get new reference results (all k, up to a certain
+      // max size)
+      auto all_size      = std::min(params_.num_db_vecs, 1024);
+      auto all_indices   = raft::make_device_matrix<int, int64_t>(handle_, num_queries, all_size);
+      auto all_distances = raft::make_device_matrix<T, int64_t>(handle_, num_queries, all_size);
+      raft::neighbors::brute_force::search<T, int>(
+        handle_, idx, query_view, all_indices.view(), all_distances.view());
+
+      int64_t offset = 0;
+      for (auto batch : batch_k_query<T, int>(handle_, idx, query_view, k_)) {
+        auto batch_size = batch.indices().extent(1);
+        auto indices    = raft::make_device_matrix<int, int64_t>(handle_, num_queries, batch_size);
+        auto distances  = raft::make_device_matrix<T, int64_t>(handle_, num_queries, batch_size);
+
+        matrix::slice_coordinates<int64_t> coords{0, offset, num_queries, offset + batch_size};
+
+        matrix::slice(handle_, raft::make_const_mdspan(all_indices.view()), indices.view(), coords);
+        matrix::slice(
+          handle_, raft::make_const_mdspan(all_distances.view()), distances.view(), coords);
+
+        ASSERT_TRUE(raft::spatial::knn::devArrMatchKnnPair(indices.data_handle(),
+                                                           batch.indices().data_handle(),
+                                                           distances.data_handle(),
+                                                           batch.distances().data_handle(),
+                                                           num_queries,
+                                                           batch_size,
+                                                           float(0.001),
+                                                           stream_,
+                                                           true));
+
+        offset += batch_size;
+        if (offset + batch_size > all_size) break;
+      }
+
+      // also test out with variable batch sizes
+      offset             = 0;
+      int64_t batch_size = k_;
+      batch_k_query<T, int> query(handle_, idx, query_view, batch_size);
+      for (auto it = query.begin(); it != query.end(); it.advance(batch_size)) {
```

Comment:
If the user is expected to invoke `advance()`, they should be able to pass in the `raft::resources` instance each time, right?